pkg/system/phase4_configuring.go

```diff
@@ -269,7 +269,22 @@ func (r *Reconciler) SetDesiredDeploymentEndpoint() error {
 	podSpec.SecurityContext.RunAsUser = &rootUIDGid
 	podSpec.SecurityContext.RunAsGroup = &rootUIDGid
 	podSpec.ServiceAccountName = "noobaa-endpoint"
-
+	honor := corev1.NodeInclusionPolicyHonor
+	disableDefaultTopologyConstraints, found := r.NooBaa.ObjectMeta.Annotations[util.DisableDefaultTopologyConstraints]
+	if (found && disableDefaultTopologyConstraints == "false") && podSpec.TopologySpreadConstraints == nil {
```

Comment:
I think we better also check here if the specific feature-gate for NodeInclusionPolicyInPodTopologySpread is enabled on this cluster. 

Also I think the conditions can be written with a bit more information:
```suggestion
	honor := corev1.NodeInclusionPolicyHonor
	hasNodeInclusionPolicyInPodTopologySpread := // ... how to get feature gate ...
	disableDefaultTopologyConstraints, found := r.NooBaa.ObjectMeta.Annotations[util.DisableDefaultTopologyConstraints]
	if podSpec.TopologySpreadConstraints != nil {
		log.Infof("deployment %s TopologySpreadConstraints already exists, leaving as is", r.DeploymentEndpoint.Name)
	} else if hasNodeInclusionPolicyInPodTopologySpread {
		log.Infof("deployment %s TopologySpreadConstraints cannot be set because feature gate NodeInclusionPolicyInPodTopologySpread is not supported on this cluster version", r.DeploymentEndpoint.Name)
	} else if found && disableDefaultTopologyConstraints == "true" {
		log.Infof("deployment %s TopologySpreadConstraints will not be set because annotation %s was set on noobaa CR", r.DeploymentEndpoint.Name, util.DisableDefaultTopologyConstraints)
	} else {
		// ... set the spec
	}
```

Reply:
@guymguym I checked the 4.14 cluster and  NodeInclusionPolicyInPodTopologySpread is not explicitly mentioned in FeatureGate CR.  Openshift team mentioned that Instead of checking the `NodeInclusionPolicyInPodTopologySpread` we can check the OCP version and every OCP/Kubernetes version until 4.12 has the feature disabled. 4.13+ have it enabled. WDYT?

Reply:
sounds ok to me